Backend/CalendarService.cs

```diff
@@ -0,0 +1,77 @@
+using FRITeam.Swapify.Backend.Interfaces;
+using FRITeam.Swapify.SwapifyBase.Entities;
+using FRITeam.Swapify.SwapifyBase.Settings;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FRITeam.Swapify.Backend
+{
+    public class CalendarService : ICalendarService
+    {
+        private readonly int dayOfWeek = 0;
+        private readonly CalendarSettings _calendarSettings;
+        private readonly DateTime startDateTime;
+        private readonly DateTime endDateTime;
+
+        public CalendarService(IOptions<CalendarSettings> calendarSettings)
+        {
+            _calendarSettings = calendarSettings.Value;
+
+            DateTime dateNow = DateTime.Now;
+            if (dateNow.CompareTo(_calendarSettings.StartWinter) == 1 && dateNow.CompareTo(_calendarSettings.EndWinter) == -1)
+            {
+                startDateTime = _calendarSettings.StartWinter;
+                endDateTime = _calendarSettings.EndWinter;
+            }
+            else
+            {
+                startDateTime = _calendarSettings.StartSummer;
+                endDateTime = _calendarSettings.EndSummer;
+            }
+            dayOfWeek = dateNow.DayOfWeek == DayOfWeek.Sunday ? 7 : (int)dateNow.DayOfWeek;
+        }
+
+        public StringBuilder StartCalendar()
+        {
+            return new StringBuilder()
+                .AppendLine("BEGIN:VCALENDAR")
+                .AppendLine("VERSION:2.0")
+                .AppendLine("CALSCALE:GREGORIAN")
+                .AppendLine("METHOD:PUBLISH")
+                .AppendLine("BEGIN:VTIMEZONE")
+                .AppendLine("TZID:Europe/Amsterdam")
+                .AppendLine("BEGIN:STANDARD")
+                .AppendLine("TZOFFSETTO:+0100")
+                .AppendLine("TZOFFSETFROM:+0100")
+                .AppendLine("END:STANDARD")
+                .AppendLine("END:VTIMEZONE");
+        }
+        public StringBuilder CreateEvent(StringBuilder stringBuilder, Block block, Course course)
+        {
+            DateTime now = DateTime.Now;
+            DateTime start = now.AddDays(-dayOfWeek + block.Day.GetHashCode())
```

Comment:
It is not clear to me what and how you calculate this start... And probably it will not be clear also to you after several weeks. May be it is a time to write some explanation comment here

Reply:
in latest commit I changed calculating of this start.. now it should be clear.